ml/src/main/scala/ai/h2o/sparkling/ml/params/AlgoParam.scala

```diff
@@ -0,0 +1,81 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package ai.h2o.sparkling.ml.params
+
+import ai.h2o.sparkling.ml.algos._
+import hex.Model
+import org.apache.spark.ml.param.{Param, ParamPair, Params}
+import org.json4s.JsonAST.{JArray, JNull, JString}
+import org.json4s.JsonDSL._
+import org.json4s._
+import org.json4s.jackson.JsonMethods.{compact, parse, render}
+
+class AlgoParam(parent: Params, name: String, doc: String, isValid: H2OSupervisedAlgorithm[_, _, _ <: Model.Parameters] => Boolean)
```

Comment:
Is the serde logic covered somewhere in unit tests?

Reply:
Good catch, wrote several tests in the H2OGridSearchTestSuite